motor/discovery/gcp/mql_assets.go

```diff
@@ -0,0 +1,174 @@
+package gcp
+
+import (
+	"errors"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/rs/zerolog/log"
+	"go.mondoo.com/cnquery"
+	"go.mondoo.com/cnquery/llx"
+	"go.mondoo.com/cnquery/motor"
+	"go.mondoo.com/cnquery/motor/asset"
+	"go.mondoo.com/cnquery/motor/discovery/common"
+	"go.mondoo.com/cnquery/motor/platform"
+	"go.mondoo.com/cnquery/motor/providers"
+	gcpprovider "go.mondoo.com/cnquery/motor/providers/google"
+	"go.mondoo.com/cnquery/mql"
+	"go.mondoo.com/cnquery/resources"
+	resource_pack "go.mondoo.com/cnquery/resources/packs/gcp"
+)
+
+const RegionLabel string = "mondoo.com/region"
+
+type MqlDiscovery struct {
+	rt *resources.Runtime
+}
+
+func NewMQLAssetsDiscovery(provider *gcpprovider.Provider) (*MqlDiscovery, error) {
+	m, err := motor.New(provider)
+	if err != nil {
+		return nil, err
+	}
+	rt := resources.NewRuntime(resource_pack.Registry, m)
+	return &MqlDiscovery{rt: rt}, nil
+}
+
+func (md *MqlDiscovery) Close() {
+	if md.rt != nil && md.rt.Motor != nil {
+		md.rt.Motor.Close()
+	}
+}
+
+func (md *MqlDiscovery) GetList(query string) []interface{} {
+	mqlExecutor := mql.New(md.rt, cnquery.DefaultFeatures)
+	value, err := mqlExecutor.Exec(query, map[string]*llx.Primitive{})
+	if err != nil {
+		return nil
+	}
+
+	a := []interface{}{}
+	d, _ := mapstructure.NewDecoder(&mapstructure.DecoderConfig{
+		Result: &a,
+	})
+	d.Decode(value.Value)
+	return a
+}
+
+func GatherMQLObjects(tc *providers.Config, project string) ([]*asset.Asset, error) {
```

Comment:
nit: this functions returns assets. Perhaps rename it to `GatherAssets` to something similar?